specs/WebNotification.md

```diff
@@ -0,0 +1,610 @@
+  /// the constructor's options parameter.
+  /// Returns `null` if the optional Notification property does not exist.
+  ///
+  /// The caller must free the returned string with `CoTaskMemFree`.  See
+  /// [API Conventions](/microsoft-edge/webview2/concepts/win32-api-conventions#strings).
+  [propget] HRESULT Icon ([out, retval] LPWSTR* icon);
+
+  /// The title of the notification as specified in the first parameter of the
+  /// constructor.
+  ///
+  /// The caller must free the returned string with `CoTaskMemFree`.  See
+  /// [API Conventions](/microsoft-edge/webview2/concepts/win32-api-conventions#strings).
+  [propget] HRESULT Title ([out, retval] LPWSTR* title);
+
+
+  /// The actions available for users to choose from for interacting with the
+  /// notification. Note that actions are only supported for persistent notifications.
+  /// Returns `null` if the optional Notification property does not exist.
+  [propget] HRESULT Actions([out, retval] ICoreWebView2NotificationActionCollection** actions);
+
+  /// The URL of the image used to represent the notification when there is not
+  /// enough space to display the notification itself.
+  /// Returns `null` if the optional Notification property does not exist.
+  ///
+  /// The caller must free the returned string with `CoTaskMemFree`.  See
+  /// [API Conventions](/microsoft-edge/webview2/concepts/win32-api-conventions#strings).
+  [propget] HRESULT Badge([out, retval] LPWSTR* badge);
+
+  /// The URL of an image to be displayed as part of the notification.
+  /// Returns `null` if the optional Notification property does not exist.
+  ///
+  /// The caller must free the returned string with `CoTaskMemFree`.  See
+  /// [API Conventions](/microsoft-edge/webview2/concepts/win32-api-conventions#strings).
+  [propget] HRESULT Image([out, retval] LPWSTR* image);
+
+  /// Specifies whether the user should be notified after a new notification
+  /// replaces an old one.
+  /// The default value is `FALSE`.
+  [propget] HRESULT Renotify([out, retval] BOOL* renotify);
+
+  /// A boolean value indicating that a notification should remain active until
+  /// the user clicks or dismisses it, rather than closing automatically.
+  /// The default value is `FALSE`.
+  [propget] HRESULT RequireInteraction([out, retval] BOOL* requireInteraction);
+
+  /// Specifies whether the notification should be silent — i.e., no sounds or
+  /// vibrations should be issued, regardless of the device settings.
+  /// The default value is `FALSE`.
+  [propget] HRESULT Silent([out, retval] BOOL* silent);
+
+  /// Specifies the time at which a notification is created or applicable (past,
+  /// present, or future).
+  /// Returns `null` if the optional Notification property does not exist.
+  [propget] HRESULT Timestamp([out, retval] double* timestamp);
+
+  /// Specifies a vibration pattern for devices with vibration hardware to emit.
+  /// Returns `null` if the optional Notification property does not exist.
```

Comment:
Must specify how to interpret this collection of unsigned long values. If this is set by the DOM API then maybe one sentence summary and a link to the spec that defines it